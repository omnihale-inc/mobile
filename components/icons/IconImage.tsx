import { Image } from "expo-image";
import { Pressable } from "react-native";

type IconImageProps = {
  onPress?: () => void;
  imageSource: () => any;
  size?: { width: number; height: number };
};

const IconImage = ({
  onPress,
  imageSource,
  size = { width: 22, height: 22 }
}: IconImageProps) => {
  return (
    // Returns the user to previous screen
    <Pressable onPress={onPress}>
      <Image
        style={{ width: size.width, height: size.height }}
        source={imageSource()}
      />
    </Pressable>
  );
};

export default IconImage;
