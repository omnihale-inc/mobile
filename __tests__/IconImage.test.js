import renderer from "react-test-renderer";

import IconImage from "../components/icons/IconImage";

it("render icon image", () => {
  const tree = renderer
    .create(<IconImage imageSource={() => "path/to/imagee"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
