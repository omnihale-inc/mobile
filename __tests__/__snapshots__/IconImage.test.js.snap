// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render icon image 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <ViewManagerAdapter_ExpoImage
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={22}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "uri": "path/to/imagee",
        },
      ]
    }
    style={
      {
        "height": 22,
        "width": 22,
      }
    }
    transition={null}
    width={22}
  />
</View>
`;
